.commitlintrc.json

```diff
@@ -0,0 +1 @@
+{ "extends": ["@commitlint/config-conventional"] }
```

Comment:
I see a lot of unnecessary/unrelated changes in this PR that should be reverted to speed up reviewing. The first 9 files are all unrelated. The only useful change maybe is this:

```diff
+            - 22
+            - 23
```